inc/utilities/class-oauth2-wp-cli.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace WP\OAuth2\Utilities;
+
+class Oauth2_Wp_Cli extends \WP_CLI_Command {
+	/**
+	* Generate a code challenge.
+	*
+	* ## OPTIONS
+	*
+	* [<random_string>]
+	* : The string to be hashed.
+	*
+	*
+	* [--length=<length>]
+	* : The length of the random seed string.
+	* ---
+	* default: 64
+	* ---
+	*
+	* ## EXAMPLES
+	*
+	*     wp oauth2 generate-code-challenge --length=64
+	*
+	* @alias generate-code-challenge
+	*/
+	function generate_code_challenge( $args, $assoc_args ) {
+		if ( ! empty( $args[0] ) && ! empty( $assoc_args['length'] ) ) {
+			\WP_CLI::warning( 'Length parameter will be ignored since the input string was provided.' );
```

Comment:
`WP_CLI` should be `use`d at the top of the file rather than absolute referenced